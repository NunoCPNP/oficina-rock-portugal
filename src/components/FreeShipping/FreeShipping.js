import useTranslation from 'next-translate/useTranslation'

import { Container } from './FreeShipping.styles'

const FreeShipping = () => {
  const { t } = useTranslation('common')

  return (
    <Container>
      <span>{t(`free-shipping`)}</span>
    </Container>
  )
}

export default FreeShipping
