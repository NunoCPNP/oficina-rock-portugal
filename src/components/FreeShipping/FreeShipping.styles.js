import styled from '@emotion/styled'

export const Container = styled.div`
  display: flex;
  justify-content: flex-end;

  span {
    margin: 0 2rem;
    padding: 0.4rem 1rem;
    border-radius: 0.5rem;
    background-color: var(--darkLiver);
    color: var(--snow);
  }
`
