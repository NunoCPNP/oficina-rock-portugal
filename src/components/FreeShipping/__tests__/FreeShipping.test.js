import { render } from '@testing-library/react'

import FreeShipping from '../FreeShipping'

describe('<FreeShipping /> spec', () => {
  it('Should match snapshot', () => {
    const { container } = render(<FreeShipping />)

    expect(container).toMatchSnapshot()
  })

  it('Should render translation keys', () => {
    const { getByText } = render(<FreeShipping />)

    expect(getByText('common:free-shipping')).toBeInTheDocument()
  })
})
