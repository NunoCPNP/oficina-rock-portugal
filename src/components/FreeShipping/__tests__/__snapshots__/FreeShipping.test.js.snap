// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FreeShipping /> spec Should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-0 span {
  margin: 0 2rem;
  padding: 0.4rem 1rem;
  border-radius: 0.5rem;
  background-color: var(--darkLiver);
  color: var(--snow);
}

<div>
  <div
    class="emotion-0"
  >
    <span>
      common:free-shipping
    </span>
  </div>
</div>
`;
